src/plugin.js

```diff
@@ -129,8 +130,13 @@ function bind (vm, fullKey, source) {
  */
 function bindAsArray (vm, fullKey, module, key, source, cancelCallback) {
   var state = vm.$store.state
+
   // set it as an array
-  utils.vuex.initWithValue(state, module, key, [])
+  vm.$store.commit(utils.vuex.getMutationName(module, VUEXFIRE_INIT_WITH_VALUE), {
+    key: key,
+    value: [],
+    module: module
+  })
 
   const onAdd = source.on('child_added', function (snapshot, prevKey) {
     const array = utils.vuex.get(state, module, key)
```

Comment:
You should actually create the array before the commit and use it as the value above